packages/core-api/__tests__/__support__/setup.ts

```diff
@@ -20,7 +21,7 @@ async function setUp() {
         ],
     });
 
-    const connection = app.resolvePlugin("database");
+    const connection = app.resolvePlugin<PostgresConnection>("database");
```

Comment:
All references to `<PostgresConnection>` should point to `<ConnectionInterface>` instead as this reference is made outside of the `core-database-postgres` package which means the `PostgresConnection` reference could be gone at any time and instead be `<SqliteConnection>` for example.

https://github.com/ArkEcosystem/core/blob/develop/packages/core-database/src/interface.ts#L14

Reply:
Yes, it'll need to be changed. There's a lot of methods and properties being referenced by their deps that are on the `PostgresConnection` class, and not the abstract class.